packages/engine/src/store/types.ts

```diff
@@ -123,15 +124,43 @@ export type SketchConfigParam =
   | SketchConfigParamVector3
   | SketchConfigParamRGB
 
-export interface SketchConfig {
+export type EnsureRequiredValueType<T> = T extends { valueType?: infer V }
+  ? Omit<T, 'valueType'> & { valueType: V }
+  : T
+
+export type SketchConfigParamImported = EnsureRequiredValueType<SketchConfigParam> & {
+  groupIndex: number | null
   title: string
-  description?: string
+  params: (SketchConfigParam | SketchConfigGroup)[]
+}
+
+export interface SketchConfigGroup {
+  groupTitle?: string
   params: SketchConfigParam[]
 }
 
+export interface SketchConfigGroupImported extends SketchConfigGroup {
+  groupTitle: string
+  groupIndex: number
+}
+
+// As the user defines the config, it can be a mix of params and groups
+export interface SketchConfigRaw {
+  title?: string
+  description?: string
+  params?: (SketchConfigParam | SketchConfigGroup)[]
+}
+
+export interface SketchConfigImported {
+  title: string
+  description?: string
+  params: SketchConfigParamImported[]
+  groupInfo: { groupTitle: string }[]
+}
+
```

Comment:
We now have types for both the raw config files that users write, and the imported version that Hedron deals with after importing the file.